lib/java/src/org/apache/thrift/TBaseAsyncProcessor.java

```diff
@@ -52,17 +52,19 @@ public boolean process(final AsyncFrameBuffer fb) throws TException {
         final TMessage msg = in.readMessageBegin();
         AsyncProcessFunction fn = processMap.get(msg.name);
         if (fn == null) {
+            TApplicationException x = new TApplicationException(TApplicationException.UNKNOWN_METHOD,
+              "Invalid method name: '" + msg.name + "'");
             TProtocolUtil.skip(in, TType.STRUCT);
             in.readMessageEnd();
-            if (!fn.isOneway()) {
-              TApplicationException x = new TApplicationException(TApplicationException.UNKNOWN_METHOD, "Invalid method name: '"+msg.name+"'");
+            // this means it is a two-way request, so we can send a reply
+            if (msg.type == TMessageType.CALL) {
               out.writeMessageBegin(new TMessage(msg.name, TMessageType.EXCEPTION, msg.seqid));
               x.write(out);
               out.writeMessageEnd();
               out.getTransport().flush();
             }
             fb.responseReady();
-            return true;
+            throw x;
```

Comment:
I don't think this is correct.  The exception is intended for the client that made the request, and that was handled.  We could certainly add an else case here and put something into the log as a debug level entry to handle oneway requests that fail... but throwing here will probably stop the server, and it should not.

Reply:
Yup.  I'm with you now.

I was looking:
https://github.com/apache/thrift/blob/75bac10dbd8d953bb2f556b7dcc3bdf7518e67a6/lib/java/src/org/apache/thrift/server/AbstractNonblockingServer.java#L606-L607

I was thinking that it would be caught here and simply logged, but I overlooked the 'return' statement.  Indeed, a Throwable will cause an exit.